contracts/savings-app/tests/interface.rs

```diff
@@ -135,15 +145,40 @@ pub fn deploy<Chain: CwEnv + Stargate>(
         .publisher_builder(Namespace::new("abstract")?)
         .build()?;
     // The dex adapter
-    publisher.publish_adapter::<_, abstract_dex_adapter::interface::DexAdapter<Chain>>(
-        abstract_dex_adapter::msg::DexInstantiateMsg {
-            swap_fee: Decimal::percent(1),
-            recipient_account: 0,
-        },
-    )?;
+    let dex_adapter = publisher
+        .publish_adapter::<_, abstract_dex_adapter::interface::DexAdapter<Chain>>(
+            abstract_dex_adapter::msg::DexInstantiateMsg {
+                swap_fee: Decimal::percent(1),
+                recipient_account: 0,
+            },
+        )?;
     // The savings app
     publisher.publish_app::<app::contract::interface::AppInterface<Chain>>()?;
 
+    let app_code = client
+        .version_control()
+        .get_app_code(APP_ID, ModuleVersion::Latest)?;
+
+    // If we create position on instantiate - give auth
+    if create_position.is_some() {
+        // TODO: We can't get account factory or module factory objects from the client.
+        // get Account id of the upcoming sub-account
+        let abs = Abstract::load_from(chain.clone())?;
+        let account_factory_config = abs.account_factory.config()?;
+        let next_local_account_id = AccountId::local(account_factory_config.local_account_sequence);
+
+        // Get salt for the module
+        let salt = generate_instantiate_salt(&next_local_account_id);
+
+        let wasm_querier = chain.wasm_querier();
+
+        let module_factory_addr = abs.module_factory.address()?;
+        let savings_app_addr = wasm_querier
+            .instantiate2_addr(app_code, module_factory_addr, salt)
+            .unwrap();
+        give_authorizations(&client, savings_app_addr)?;
```

Comment:
How can we expose it nicely for abstract-client? We can expose `next_local_account_id` as well as `module_factory_address` for the client if we're not planning to expose factory objects

Reply:
Created an issue! https://linear.app/abstract-sdk/issue/ABS-316/implement-a-module-instantiate2-address-getter